using-multifactor-authentication.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: default
+title: Using multi-factor authentication
+url: /using-multifactor-authentication
+previous: /run-your-own-gem-server
+next: /resources
+---
+
+<em class="t-gray">Want to better protect your RubyGems account?</em>
+
+Your account of RubyGems is important, just like your other accounts. Many
+websites now has been using an extra phase of user authentication (MFA), to
+protect data of users better. Now, you can also have a taste of multi-factor
+authentication on RubyGems.org.
+
+Since this feature is still in preview process, it's deployed but hidden by
+default. You can visit [RubyGems.org](https://rubygems.org), open JavaScript
+console in your browser, and type following command.
+
+    document.cookie = 'mfa_feature=true;path=/'
+
+This cookie entry will make the server aware of your request to try this
+feature.
+
+## Prerequisite
+
+You should have an authenticator app (like Google Authenticator or Authy) which
+supports time-based one-time password (TOTP) to scan the QR-code and generate
+digit code. SMS-based authentication or recovery is not supported.
+
+## Enabling multi-factor authentication
+
+Login at the site using your account and enter _edit profile_ page. You can see
+a section named _multifactor authentication_. Click _register a new device_
+button to continue.
+
+Now you are directed to a page with QR-code and an text box for verifying digit
+code. Use your authenticator to scan the QR-code. A new item will be shown as
+soon as the scan succeeds. If your authenticator cannot scan from camera, you
+can choose to manually add a new item. Fill information listed after the
+QR-code. Be cautious to type the _key_ correctly and choose _time based_
+option, and click _save_. You can see a 6-digit code changing with time (30
+seconds) in your app. Type the code and click _enable_.
+
+If the code is correct and QR-code not expired, you can see a list of recovery
+codes. Please keep them safe. Any of these codes can be used in place of digit
+code generated from authenticator, but every one can only be used once. So if
+your phone is missing or you unexpectedly deleted the account in the
+authenticator, you can use recovery code to disable multi-factor authentication
+first and re-generate a new key.
```

Comment:
@ecnelises I think a numbered list of steps to take with appropriate screenshots would work better here. Something similar to https://docs.microsoft.com/en-us/azure/active-directory/user-help/microsoft-authenticator-app-how-to#add-a-work-or-school-account